protos/NUgus.proto

```diff
@@ -1297,21 +1305,55 @@ PROTO NUgus [
                     translation 0.000000 0.000000 0.000000
                     rotation 0.000000 0.000000 1.000000 -1.570790
                     children [
-                      Camera {
+                      DEF RIGHT_CAMERA Camera {
                         name "right_camera"
-                        translation 0.069952 -0.033795 0.06488
-                        rotation -0.570515 0.570516 0.590782 -2.07437
-                        fieldOfView 1.5707
-                        width 640
-                        height 480
+                        translation 0.069952 -0.033795 0.06488          # right side of the face
+                        rotation -0.570515 0.570516 0.590782 -2.07437   # rotated up in y by 2 degrees
+                        
+                        # Set parameters from fields
+                        fieldOfView IS fieldOfView
+                        width IS cameraWidth
+                        height IS cameraHeight
+                        noise IS cameraNoise
+                        motionBlur IS cameraMotionBlur
+                        
+                        # We are using a rectilinear lens because there is no true spherical lens
+                        spherical FALSE
+                        lens DEF LENS Lens {
+                          # Parameters are set to default
+                          center 0.5 0.5
+                          radialCoefficients 0 0
+                          tangentialCoefficients 0 0
+                        }
+                        focus DEF FOCUS Focus {
```

Comment:
@wongjoel and I were thinking of removing the Focus node for these reasons:

- It's directly related to FOV, and setting one should set the other. Setting both could possibly lead to conflicts/unwanted behavior? We don't know. Also, we're calculating the focal length dynamically based on the other parameters (in the vision data collection).
- Using a Focus node allows for changing the focus at run time, but our real lens have their focus pretty much fixed (e.g. they're preset based on how much the lens are screwed in).

I've tried running without the Focus node, and I couldn't see any difference. We can keep it though if its required for model inspection, and make sure we always set a value that doesn't conflict with the other related parameters.

Reply:
Doesn't look like it's required for model inspection. If it's not there, it's just set to the default values and these are the default values, so there won't be any difference. This is the same for the lens parameters too.